runner_manager/models/runner_group.py

```diff
@@ -162,7 +166,7 @@ def need_new_runner(self) -> bool:
         runners = self.get_runners()
         not_active = len([runner for runner in runners if runner.is_active is False])
         count = len(runners)
-        return not_active < self.min and count < self.max
+        return (not_active < self.min and count < self.max) or self.queued > 0
```

Comment:
if we reach max but we have a queued runner we don't want to return `True` but `False`.

```suggestion
        return (not_active < self.min or self.queued > 0) and count < self.max
```

Not so sure about my suggestion so please create a testing scenario that has queued runners and max is == count